### PR TITLE
Feat/7 dictionary diff

### PR DESCRIPTION
**Pull request by andricDu**

### CODE REVIEW

File: src/controllers/dictionaryController.ts [hunk 1]
```diff
@@ -40,3 +42,18 @@ export const updateFile = (req: Request, res: Response, next: NextFunction) => {
         res.status(200).send(dict.toObject());
     }).catch(next);
 };
+
+export const diffDictionaries = async (req: Request, res: Response, next: NextFunction) => {
```
Comment (blabadi) [inline src/controllers/dictionaryController.ts]: this shouldn't be async like other handlers you have or the app will crash

**Comment by andricDu** [comment]
Decorated the handlers with the asyncWrapper

File: package.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
     "serve": "node dist/server.js",
     "watch-node": "nodemon dist/server.js",
     "test": "mocha --timeout 35000 --exit -r ts-node/register test/**/*.ts",
+    "watch-test": "mocha --watch --timeout 35000 --exit -r ts-node/register test/**/*.ts",
```
[the remark this answers is not in the document]
Comment (andricDu) [reply package.json]: Doesn't work correctly for now, may be an issue with the latest version of mocha.

File: src/controllers/dictionaryController.ts [hunk 3]
```diff
@@ -16,27 +16,38 @@ export const listDictionaries = async (req: Request, res: Response) => {
...
+    const leftVersion = req.query.left;
+    const rightVersion = req.query.right;
+
+    if (name && leftVersion && rightVersion) {
+        const dict1 = await dictionaryService.findOne(name, leftVersion);
+        const dict2 = await dictionaryService.findOne(name, rightVersion);
+        const diff = diffUtil(dict1.toObject(), dict2.toObject());
+        res.status(200).send(Array.from(diff.entries()));
+    } else {
+        next(new BadRequestError("name and left and right versions must be set"));
```
Comment (blabadi) [inline src/controllers/dictionaryController.ts]: will this still work ? I assume it will

**Pull request merged**